src/ui/ui_i18n.js

```diff
@@ -0,0 +1,43 @@
+import langParser from 'accept-language-parser';
+import _ from 'lodash';
+
+/**
+ * Return translations suitable from a user side locale list, substituting any missing
+ * translation with the default local translation.
+ * @param {String} acceptLanguages - Accept languages/locales from user side
+ * @param {Object} server - Hapi server instance
+ * @return {Object} - A JSON object of translations
+ */
+export async function getTranslations(acceptLanguages, server) {
+  const languageTags = getLanguageTags(acceptLanguages);
+  const translations = await server.plugins.i18n.getTranslationsForLocales(languageTags);
+  let fullTranslations = {};
+  if (_.isEmpty(translations)) {
+    fullTranslations = await server.plugins.i18n.getTranslationsForDefaultLocale(server);
+  } else {
+    const defaultTranslations = await server.plugins.i18n.getTranslationsForDefaultLocale(server);
+    fullTranslations = _.defaults(translations, defaultTranslations);
+  }
+  return fullTranslations;
+};
```

Comment:
Looking at it now and how it has been refactored, I agree with your assessment. I have removed the module and its corresponding test module.